MergerLogic/ImageProcessing/TileMerger.cs

```diff
@@ -62,7 +74,7 @@ public TileMerger(ITileScaler tileScaler, ILogger<TileMerger> logger)
             return tile;
         }
 
-        private List<MagickImage> GetImageList(List<CorrespondingTileBuilder> tiles, Coord targetCoords)
+        private List<MagickImage> GetImageList(List<CorrespondingTileBuilder> tiles, Coord targetCoords, bool uploadOnly)
```

Comment:
i understand the need to use "uploadOnly" flag to prevent unneccassery call to destination, where - no tiles where fond in sources and uploadOnly is set to "false" but i do think it's ugly to use this flag when you removed the "destination" tile from tiles array earlier - can't we think on better refactor?